admin/digilan-admin.php

```diff
@@ -417,6 +424,135 @@ private static function updateCityscopeCloud($cityscopeCloud)
         exit();
     }
 
+    private static function sanitize_form_field_option_by_type($field_option, $unsafe_value)
+    {
+        if ($unsafe_value === '') {
+            return $unsafe_value;
+        }
+        switch ($field_option) {
+            case 'type':
+                return DigilanTokenSanitize::sanitize_form_field_type($unsafe_value);
+                break;
+            case 'display-name':
+                return DigilanTokenSanitize::sanitize_form_field_display_name($unsafe_value);
+                break;
+            case 'instruction':
+                return DigilanTokenSanitize::sanitize_form_field_instruction($unsafe_value);
+                break;
+            case 'unit':
+                return DigilanTokenSanitize::sanitize_form_field_unit($unsafe_value);
+                break;
+            case 'options':
+                return DigilanTokenSanitize::sanitize_form_field_options($unsafe_value);
+                break;
+            default:
+                _default_wp_die_handler(sprintf('Unhandled field option: %s', $field_key));
+                break;
+        }
+    }
+
+    private static function add_field_to_form()
+    {
+        $user_form_fields = get_option('digilan_token_user_form_fields');
+        if (false === $user_form_fields ) {
+            add_option("digilan_token_user_form_fields", array());
+        }
+        $new_field_data = array();
+        foreach($_POST as $post_key => $post_value) {
+            // $post_key example : 
+            // - dlt-custom-portal-new-field/display-name/fr_FR
+            // - dlt-custom-portal-new-field/type
+            // - action
+            [$prefix, $field_option, $lang_code] = explode('/', $post_key);
+            if ($prefix !== 'digilan-token-new-field') {
+                continue;
+            }
+
+            $field_value = self::sanitize_form_field_option_by_type($field_option, $post_value);
+            if (false === $field_value) {
+                \DLT\Notices::addError(sprintf('Invalid value for %s', $field_option));
+                wp_redirect(self::getAdminUrl('form-settings'));
+                exit();
+            }
+            if ($lang_code) {
+                // handle field options with translations (displayed to user)
+                $new_field_data[$field_option][$lang_code] = $field_value;
+            } else {
+                // handle hidden field options without translations (like input type)
+                $new_field_data[$field_option] = $field_value;
+            }
+        }
+        $new_field_data_filtered = array_filter($new_field_data, function($data) {
+            return is_array($data) ? array_filter($data) : $data;
+        });
```

Comment:
array_filter
avec un seul argument ?
`If no callback is supplied, all empty entries of array will be removed. See [empty()](https://www.php.net/manual/en/function.empty.php) for how PHP defines empty in this case.``

donc tu essaie d enlever des null dans des sous tableaux ?

Ces sous tableaux existerais pas si tu avais suivi mes reco précédente: 

langue par default et pas d iteration.

Tout cela est compliqué pour pas grand chose et dur à valider. 

Par ailleurs , les commentaires au dessus aide pas du tout 
`// handle hidden field options without translations (like input type)`

oui on avait compris que y avais pas de langue.... mais cela handle rien

par contre quand tu écris 
``` 
 $new_field_data_filtered = array_filter($new_field_data, function($data) {
            return is_array($data) ? array_filter($data) : $data;
        });
```
pour 
// delete empty value in subarrays

en faisant 4 appel externe sur deux lignes :-/ pour des truc que tu viens de créer c est vraiment tres tres moyen

encore une fois l iteration est pas une bonne idée

Ta liste de post devrait etre plus structuré tu recois genre la liste des champs dans un truc , tu verifies sui leur nom est valide puis tu itere la dessus pour extraire leurs infos.

Aussi des structures dynamiques à type variable c'est trop compliqué , créer la langue DEFAULT ou choisie une langue default et drop le cas pas de langue merci